init.rb

```diff
@@ -52,93 +52,58 @@
     Autoproj.gitorious_server_configuration('GITHUB', 'github.com', :http_url => 'https://github.com')
 end
 
-require File.join(__dir__, 'rock/flavor_definition')
-require File.join(__dir__, 'rock/flavor_manager')
-require File.join(__dir__, 'rock/in_flavor_context')
 require File.join(__dir__, 'rock/current_release')
 require File.join(__dir__, 'rock/python')
 
-Rock.flavors.define 'stable'
-Rock.flavors.alias 'stable', 'next'
-Rock.flavors.define 'master', :implicit => true
-
-configuration_option('ROCK_SELECTED_FLAVOR', 'string',
-    :default => 'master',
-    :possible_values => ['stable', 'master'],
-    :doc => [
-        "Which flavor of Rock do you want to use ?",
-        "Stay with the default ('master') if you want to use Rock on the most recent",
-        "distributions (Ubuntu 16.04 and later). Use 'stable' only for ",
-        "now officially unsupported distributions (Ubuntu 14.04)"])
-
-if Rock.in_release? && !Autoproj.config.has_value_for?('ROCK_SELECTED_FLAVOR')
-    Autoproj.config.set 'ROCK_SELECTED_FLAVOR', 'stable', true
-end
-
-Rock.flavors.select_current_flavor_by_name(
-    ENV['ROCK_FORCE_FLAVOR'] || Autoproj.config.get('ROCK_SELECTED_FLAVOR'))
-
-current_flavor = Rock.flavors.current_flavor
-
-#This check is needed because the overrides file will override the FLAVOR selection.
-#Furthermore a selection != stable can cause a inconsistent layout (cause by in_flavor system in the package_sets)
-if Rock.in_release? && current_flavor.branch != "stable" 
-    if ENV['ROCK_RC'] == '1'
-        Autoproj.warn ""
-        Autoproj.warn "Found a release file and the flavor is not master"
-        Autoproj.warn "This would usually be an error, but since ROCK_RC is set to 1,"
-        Autoproj.warn "it is assumed that you're preparing a release candidate"
-        Autoproj.warn ""
-    else
-        Autoproj.error ""
-        Autoproj.error "You selected the flavor '#{current_flavor.branch}' but '#{File.join(Autoproj.root_dir,"autoproj", "overrides.d", "25-release.yml")}' exists."
-        Autoproj.error "This means you are on a release; either unselect the release by calling 'rock-release switch master'"
-        Autoproj.error "or call 'autoproj reconfigure' and select the FLAVOR 'stable'"
-        exit 1
-    end
-end
 
-Autoproj.config.set('ROCK_SELECTED_FLAVOR', current_flavor.name, true)
-Autoproj.config.set('ROCK_FLAVOR', current_flavor.branch, true)
-Autoproj.config.set('ROCK_BRANCH', current_flavor.branch, true)
-
-if current_flavor.name != 'master' && Autoproj::PackageSet.respond_to?(:add_source_file)
-    Autoproj::PackageSet.add_source_file "source-stable.yml"
+# backward compatibility for the deprecated flavor system
+if !Autoproj.config.has_value_for?('ROCK_SELECTED_FLAVOR') then
```

Comment:
`then` is not idiomatic Ruby ... please remove

Reply:
If the value exists and is not 'master', I think one should raise an exception. And then unconditionally set ROCK_SELECTED_FLAVOR to master. Same for ROCK_FLAVOR. I would leave the user's choice for ROCK_BRANCH though.